crates/bevy_reflect/src/path.rs

```diff
@@ -136,12 +141,48 @@ impl GetPath for dyn Reflect {
                                     index: current_index,
                                 })
                             }
+                        },
+                        Some(Token::SingleQuote) => {
+                            let ident =
+                                if let Some(Token::Ident(ident)) = next_token(path, &mut index) {
+                                    ident
+                                } else {
+                                    return Err(ReflectPathError::ExpectedIdent {
+                                        index: current_index,
+                                    });
+                                };
+
+                            if let Some(Token::SingleQuote) = next_token(path, &mut index) {
+                            } else {
+                                return Err(ReflectPathError::ExpectedToken {
+                                    index: current_index,
+                                    token: "'",
+                                });
+                            }
```

Comment:
and `if/else` branch with one of the branches being empty is a little odd. This could be condensed to
```rust
if !matches!(Some(Token::SingleQuote), next_token(path, &mut index)) {
    return Err(..);
}
```

Reply:
You are correct, that read better indeed. I just used the empty if-branch because it was already used further down in the function to check for the CloseBracked token. I changed all cases to `if !matched!` now to keep them consistent